code/modules/clothing/neck/collar_bomb.dm

```diff
@@ -0,0 +1,120 @@
+	. = ..()
+	src.button = button
+	button?.collar = src
+	set_wires(new /datum/wires/collar_bomb(src))
+
+/obj/item/clothing/neck/collar_bomb/Destroy()
+	button?.collar = null
+	button = null
+	return ..()
+
+/obj/item/clothing/neck/collar_bomb/examine(mob/user)
+	. = ..()
+	if(user.get_item_by_slot(ITEM_SLOT_NECK) == src)
+		return
+	. += span_tinynotice("It has a [EXAMINE_HINT("wire")] panel that could be interacted with...")
+
+/obj/item/clothing/neck/collar_bomb/attackby(obj/item/item, mob/user, params)
+	if(is_wire_tool(item))
+		wires.interact(user)
+	else
+		return ..()
+
+/obj/item/clothing/neck/collar_bomb/equipped(mob/user, slot, initial = FALSE)
+	. = ..()
+	if(slot == ITEM_SLOT_NECK)
+		ADD_TRAIT(src, TRAIT_NODROP, INNATE_TRAIT)
+
+/obj/item/clothing/neck/collar_bomb/dropped(mob/user, silent = FALSE)
+	. = ..()
+	REMOVE_TRAIT(src, TRAIT_NODROP, INNATE_TRAIT)
+
+/obj/item/clothing/neck/collar_bomb/proc/explosive_countdown(ticks_left)
+	active = TRUE
+	if(ticks_left > 0)
+		playsound(src, 'sound/items/timer.ogg', 30, FALSE)
+		balloon_alert_to_viewers("[ticks_left]")
+		ticks_left--
+		addtimer(CALLBACK(src, PROC_REF(explosive_countdown), ticks_left), 1 SECONDS)
+		return
+
+	playsound(src, 'sound/effects/snap.ogg', 75, TRUE)
+	if(!ishuman(loc))
+		balloon_alert_to_viewers("dud...")
+		active = FALSE
+		return
+	var/mob/living/carbon/human/brian = loc
+	if(brian.get_item_by_slot(ITEM_SLOT_NECK) != src)
+		balloon_alert_to_viewers("dud...")
+		active = FALSE
+		return
+	balloon_alert_to_viewers(UNLINT("BOOM!!"))
+	visible_message(span_warning("[src] goes off, outright decapitating [brian]!"), span_hear("You hear a fleshy boom!"))
+	playsound(src, SFX_EXPLOSION, 30, TRUE)
+	brian.apply_damage(200, BRUTE, BODY_ZONE_HEAD)
+	var/obj/item/bodypart/head/myhead = brian.get_bodypart(BODY_ZONE_HEAD)
+	myhead?.dismember()
+	brian.investigate_log("has been decapitated by [src].", INVESTIGATE_DEATHS)
+	flash_color(brian, flash_color = "#FF0000", flash_time = 1 SECONDS)
+	qdel(src)
+
+///The button that detonates the collar.
+/obj/item/collar_bomb_button
+	name = "big yellow button"
+	desc = "It looks like a big red button, except it's yellow. It comes with a heavy trigger, to avoid accidents."
+	icon = 'icons/obj/devices/assemblies.dmi'
+	icon_state = "bigyellow"
+	inhand_icon_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/items/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items/devices_righthand.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	///The collar bomb it's associated with.
+	var/obj/item/clothing/neck/collar_bomb/collar
+
+/obj/item/collar_bomb_button/attack_self(mob/user)
+	. = ..()
+	if(DOING_INTERACTION_WITH_TARGET(user, src))
+		return
+	balloon_alert_to_viewers("pushing the button...")
+	if(!do_after(user, 1.2 SECONDS, target = src))
+		return
+	playsound(user, 'sound/machines/click.ogg', 25, TRUE)
+	balloon_alert_to_viewers("button pushed")
```

Comment:
Similarly as there's only a 1.2 second delay, I don't think this second alert is necessary (the first will probably still be visible?). 
We have a sound, and otherwise they'll likely soon hear about the result.